include/seqan3/range/view/take.hpp

```diff
@@ -117,31 +117,33 @@ class view_take : public ranges::view_interface<view_take<urng_t, exactly, or_th
          * \brief seqan3::detail::inherited_iterator_base operators are used unless specialised here.
          * \{
          */
-        iterator_type & operator++() noexcept(noexcept(++base_base_t{}))
+        iterator_type & operator++() noexcept(noexcept(++std::declval<base_t &>()))
         {
-            base_base_t::operator++();
+            base_t::operator++();
             ++pos;
             return *this;
         }
 
-        iterator_type operator++(int) noexcept(noexcept(++base_base_t{}))
+        iterator_type operator++(int) noexcept(noexcept(++std::declval<iterator_type &>()) &&
+                                               noexcept(iterator_type{std::declval<iterator_type&>()}))
```

Comment:
Not so important but wouldn't `is_nothrow_copy_constructible` be also working here? 
In that case it would be more readable, but honestly I don't really care about this, so you can decide if you want to change this.

Reply:
indeed